tm-monitor/docs/Design-proposal.md

```diff
@@ -0,0 +1,53 @@
+# Design proposal for Tendermint Monitor
+
+Tendermint monitor is designed to be the tool for all related parties to have a close look about the network. 
+
+## Design Goal for monitoring the network
+
+The design goal of Tendermint Monitor is to track **Liveness** & **Safety** of the whole network and certain validator.
+
+ The liveness of the network is in jeopardy if:
+
+* many validators are offline
+* the system barely gather +2/3 votes in recent blocks
+* validators have to wait for many rounds before commiting new block
+
+
+The safety of the network is in jeopardy if:
+
+* There exists a monopoly of Atom token, +1/3
+
+* Not enough Atoms are bonded
+
+* Certain validator attracts too many delegation
+
+
+## Design Goal for monitoring a validator
+
+The validator must monitor its **availability** and **safety** closely.  There are some early-stage indicators to be noticed:
+
+*  Are most of previous precommits included in blocks?
+* How many proposals it missed before?
+
+
+The safety of a validator is under pressure if :
+
+* it's not connected with enough persistent peers 
+* there is not enought peers in its address book
+
+## Tendermint Monitor Atchitecture
+
+The architecture of new version of monitor will be like this:
+
+![WechatIMG114](https://github.com/bianjieai/tendermint-tools/raw/bianjie/tm-monitor/tm-monitor/docs/pics/WechatIMG114.jpeg)
```

Comment:
What `Query` is? Do you mean `ABCIQuery`? If yes, then it will also go through Tendermint, not gaia.

Reply:
@melekes, yes, all the queries will go through Tendermint Core. I have updated the design pic.